### PR TITLE
[FIX] l10n_all: several fixes and normalizations for the localization tests

### PR DESCRIPTION
**Pull request by lordkrandel**
After making country_id mandatory in account_tax_groups, several tests started to fail and needed adaptation.
Also test data needed some mending.

*    account: Fixing AccountTestInvoicingCommon functions' default parameters
*    l10n_ch: Swiss qr tests no more rely on demo data
*    l10n_ar: removed the country on tax_group_taxes, which is the default group
*    l10n_co: added missing tax groups to tax templates
*    l10n_*: Assign country to tax groups declared in data files
*    l10n_*: Localization tax group localization data normalization
*    account: Tests now include country_id in the armageddon tax

### CODE REVIEW

File: addons/l10n_au/data/account.tax.group.csv [hunk 1]
```diff
@@ -1,4 +1,3 @@
-id,name
-tax_group_gst_0,GST 0%
-tax_group_gst_10,GST 10%
-tax_group_gst_100000000,GST 100000000%
```
Comment (jco-odoo) [inline addons/l10n_au/data/account.tax.group.csv:4]: Who told you to remove this?

Comment (lordkrandel) [reply addons/l10n_au/data/account.tax.group.csv:4]: You mean this? --> 
immagine
It's coming from @tsb-odoo , @oco-odoo 

**Benj:** I've been discussing this with OCO
It is a false good idea
Our suggestion is to remove that rate of 1000000... %
in the standard AU localization, let's remove it.

**Paolo:** Aaaand... someone will complain :smile:

**Benj:** The user can still : 
- create its invoice 
- go to tab Journal Items
- set the tax grid manually
- post its invoice
But the logic of 1000000...% and tax price include is a hack, not something we want to promote in standard.

**Paolo:** ooook

Comment (william-andre) [reply addons/l10n_au/data/account.tax.group.csv:4]: We need to ask them if the workaround works for them. Not just remove it without knowing.

Comment (jco-odoo) [reply addons/l10n_au/data/account.tax.group.csv:4]: @richard-willdooit

Comment (tsb-odoo) [reply addons/l10n_au/data/account.tax.group.csv:4]: Other work around identified in the meantime. 
Tax with 0% amount
Only a base line in the tax repartition lines. No tax repartition
Tax grids specified on the Base Repartition Line
Then on the vendor bill, the account to be used is not an expense account but a tax paid account.

Comment (lordkrandel) [reply addons/l10n_au/data/account.tax.group.csv:4]: Change moved to https://github.com/odoo/odoo/pull/73890, see task  https://www.odoo.com/web#id=2278230&model=project.task

**Comment by oco-odoo** [comment]
@lordkrandel could you squash your first commit and las' one ? Essentially, it'll render your first commit useless (you can keep the replacement of [] to None in the default parameter of the helper if you want, but adapt the commit message). 

Also, please respect the guidelines for you other commit messages: 

[FIX] Localization LU tax group configuration fix

should for example be:

[FIX] l10n_lu:  assign correct country to tax groups


Instead of having so many commits doing the same thing in different places, you can just put them all in a single commit named something like:

[FIX] l10n_*: assign correct country to tax groups declared in csv

**Comment by lordkrandel** [comment]
> @lordkrandel could you squash your first commit and las' one ? Essentially, it'll render your first commit useless (you can keep the replacement of [] to None in the default parameter of the helper if you want, but adapt the commit message).
Ok.

> Instead of having so many commits doing the same thing in different places, you can just put them all in a single commit named something like:
> [FIX] l10n_*: assign correct country to tax groups declared in csv
Oh, yes, I will.
I separated the several commits just to make it more readable to the reviewer.
I was thinking of squashing everything after. That's why I didn't care much about the messages.

**Review by oco-odoo — CHANGES_REQUESTED** [review]

File: addons/l10n_pe/data/account_tax_group.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            <field name="name">OTROS</field>
+            <field name="sequence">0</field>
+            <field name="country_id" ref="base.pe"/>
+        </record>
+        <record id="tax_group_det" model="account.tax.group">
+            <field name="name">DET</field>
+            <field name="sequence">100</field>
+            <field name="country_id" ref="base.pe"/>
+        </record>
+        <record id="tax_group_icbper" model="account.tax.group">
```
[the remark this answers is not in the document]
Comment (oco-odoo) [reply addons/l10n_pe/data/account_tax_group.xml:48]: So, to be 100% consistent with our stable policy, I'd do that in master in a distinct PR, with a migration script.

File: addons/l10n_ar/data/account_tax_group.xml [hunk 3]
```diff
@@ -59,7 +59,7 @@
 
     <!-- Others Taxes -->
 
-    <record model="account.tax.group" id="account.tax_group_taxes">
+    <record model="account.tax.group" id="tax_group_otros_impuestos">
         <field name="name">Otros Impuestos</field>
         <field name="sequence">20</field>
```
Comment (jco-odoo) [inline addons/l10n_ar/data/account_tax_group.xml:64]: @lordkrandel You are just renaming here.  And I don't see the point in removing the country on the tax group (which is not done here)

Comment (lordkrandel) [reply addons/l10n_ar/data/account_tax_group.xml:64]: Argentina is the only country that it's renaming and putting the country on `account.tax_group_taxes`, which is used as a default here:
https://github.com/odoo/odoo/blob/0ca0fdcee3d4610bcb14a0960dad8ad50f63884d/addons/account/models/account_tax.py#L68-L69

and that messes things up in the tests here when you have Argentina and another localization installed: 
https://github.com/odoo/odoo/blob/0ca0fdcee3d4610bcb14a0960dad8ad50f63884d/addons/account/models/account_tax.py#L130-L134

This otros_impuestos group is used only in demos in the code, I think, so I thought it was better to have a dedicated record for it and leave account.tax_group_taxes alone. 🤷🏻

Comment (oco-odoo) [reply addons/l10n_ar/data/account_tax_group.xml:64]: I agree with @lordkrandel

**Review by oco-odoo — CHANGES_REQUESTED** [review]

File: addons/account/tests/common.py [hunk 4]
```diff
@@ -294,11 +295,13 @@ def setup_armageddon_tax(cls, tax_name, company_data):
             'name': '%s (group)' % tax_name,
             'amount_type': 'group',
             'amount': 0.0,
+            'country_id': company_data['country'].id,
```
[the remark this answers is not in the document]
Comment (lordkrandel) [reply addons/account/tests/common.py:298]: I added it with the country coming from the chart_template (which if I remember correctly is the one which is used to fill out the fiscal country)
Do you want me to change the key name from `country` to `fiscal_country` ?

https://github.com/odoo/odoo/pull/73619/files/205126bfd9d00e619f379b8c8c1fe8db9b2d3a32..bc485f18506f0b4120750ecdb6c95fc1581bb42c#diff-8f9137a4cc5f642b87d839ac0afe2bd2ad49cdacbd63f77f3126157683302fc2
immagine

File: addons/l10n_co/data/account_tax_template.xml [hunk 5]
```diff
@@ -8,6 +8,7 @@
         <field name="amount_type">percent</field>
         <field name="type_tax_use">purchase</field>
         <field name="chart_template_id" ref="l10n_co_chart_template_generic"/>
+        <field name="tax_group_id" ref="tax_group_iva_5"/>
```
[the remark this answers is not in the document]
Comment (lordkrandel) [reply addons/l10n_co/data/account_tax_template.xml:11]: Done, https://github.com/odoo/odoo/pull/73619/commits/2e1340864f106f69d2b576d634f1e8e8c4551ee6

File: addons/account/tests/common.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (oco-odoo) [reply addons/account/tests/common.py:298]: Haha right XD But I think you can simply write company_data['company'].account_fiscal_country_id instead and avoid adding a key in the dict.

**Review by oco-odoo — APPROVED** [review]
@robodoo r+ rebase-ff

**Pull request closed without merging**